### ISSUE TITLE
Type of error, passed from retry filter is type unknown

### ISSUE THREAD

**Issue by StanislavMyakishev** [igorkamyshev/farfetched#260]
Hi! 
I use `createJsonQuery` with `retry` and now the error object passed to the `filter` is type `unknown`, but it looks like it should be narrowed to some error type 

```typescript
retry(basicQuery, {
  ...
  filter: {
    source: $filter,
    fn: ({ error }) => {
      // error: unknown
      return isNetworkError({ error });
    }
  }
});
```
Same `unknown` type appears if you use just `filter` handler instead of `{ source: ..., fn: ... }`

It's not blocking or anything because setting the generic type of error through the query creation works just fine and returns
`error: AbortError | TimeoutError | PreparationError | NetworkError | HttpError<number> | InvalidDataError | CustomError`
Though it would be nice to have this long error enumeration type without generics being set.

Here is the [sandbox link](https://codesandbox.io/s/sharp-breeze-fylp1j?file=/src/App.tsx), so you can check it out

Tysm for the lib, just added it to the project but enjoy it already ❤️

**Comment by igorkamyshev** [comment]
Thanks for report 🙏 I'll take a look at this bug as soon as possible, but before 😉 do you want to fix it by yourself?

**Comment by igorkamyshev** [comment]
fixed in #265, will be released in the next release
